WordPress/gutenberg
# Try better block labels

## Description
Related #17519, #18014

Introduces a new `__experimentalLabel( attributes, context )` function to the block api. The aim of this is to provide more context about a particular block which can be currently surfaced in two ways:

1. Instead of displaying the Block Title in the Block Navigator, it displays some content from the block. This is currently being used by the Navigation Block to let users differentiate their nav links:
![Screen Shot 2019-12-11 at 3 38 30 pm](https://user-images.githubusercontent.com/677833/70601105-4f222d00-1c2c-11ea-9e86-ec5781ceab46.png)

2. Providing more contextual information to screen readers, this might be paragraph text or an image caption so that users can be more sure of the block content.
![Screen Shot 2019-12-11 at 4 09 18 pm](https://user-images.githubusercontent.com/677833/70603041-b6da7700-1c30-11ea-890d-9dd848556f2f.png)

The new `__experimentalLabel` also replaces the mobile function `__experimentalGetAccessibilityLabel` and those labels now work across both mobile and web.

The blocks package now also has two new convenience functions for retrieving these labels:
- `__experimentalGetBlockLabel` - returns just the output of the `__experimentalLabel`.
- `__experimentalGetAccessibleBlockLabel` - returns some extra information useful to screen readers like the block title and the position in the block list.

Those functions will also strip any html (i.e. from formatting) from the labels.

## How has this been tested?
Additional unit testing.

Manual testing:
1. Add a Navigation Block (make sure it's enabled as an experimental feature).
2. Ensure the navigation block has menu items.
3. View the Block navigator.
4. Observe the block labels for menu items
5. Inspect the block mover for a menu item using dev tools
6. Observe the screen reader text includes the new block label
7. Inspect the menu item's block wrapper
8. Observe the aria label includes the new block label

Other blocks worth testing for the aria-label - Paragraph, Heading, More, Image.




## Types of changes


New feature (non-breaking change which adds functionality)


## Checklist:
